benchmarks/src/main/kotlin/macrobenchmarks/ChannelProducerConsumerMonteCarloBenchmark.kt

```diff
@@ -0,0 +1,264 @@
+                          generateEta : () -> String) {
+    val runExecutionTimesMs = ArrayList<Long>()
+    var lastMean = -10000.0
+    var runIteration = 0
+    while (true) {
+        repeat(ITERATIONS_BETWEEN_THRESHOLD_CHECK) {
+            runIteration++
+            runExecutionTimesMs += runIteration(threads, channel, withBalancing, withSelect, dispatcherType)
+            val result = runExecutionTimesMs.average().toInt()
+            val std = runExecutionTimesMs.standardDeviation().toInt()
+            val eta = generateEta.invoke()
+            print("\rchannel=$channel threads=$threads withBalancing=$withBalancing dispatcherType=$dispatcherType withSelect=$withSelect iteration=$runIteration result=$result std=$std [$eta]")
+        }
+        val curMean = runExecutionTimesMs.average()
+        if (runIteration >= MAX_ITERATIONS || abs(curMean - lastMean) / curMean < MONTECARLO_STOP_THRESHOLD) break
+        lastMean = curMean
+    }
+
+    val result = runExecutionTimesMs.average().toInt()
+    val std = runExecutionTimesMs.standardDeviation().toInt()
+    println("\rchannel=$channel threads=$threads withBalancing=$withBalancing dispatcherType=$dispatcherType withSelect=$withSelect result=$result std=$std iterations=$runIteration")
+
+    writeIterationResults(channel = channel, threads = threads, withBalancing = withBalancing, dispatcherType = dispatcherType,
+            withSelect = withSelect, result = result, std = std, runIteration = runIteration)
+}
+
+/**
+ * Estimated time of arrival
+ */
+fun eta(curIteration: Int, totalIterations: Int, startTime: Long): String {
+    if (curIteration == 0) return "ETA - NaN"
+    val time = System.currentTimeMillis() - startTime
+    var eta = (time.toDouble() / curIteration * totalIterations / 60_000).toInt() // in minutes
+    val minutes = eta % 60
+    eta /= 60
+    val hours = eta % 24
+    eta /= 24
+    val days = eta
+    return "ETA - $days days, $hours hours,  $minutes minutes"
+}
+
+fun runIteration(threads: Int, channelCreator: ChannelCreator, withBalancing: Boolean, withSelect: Boolean,
+                 dispatcherCreator: DispatcherCreator): Long {
+    val producers = Random.nextInt(1, max(threads, 2))
+    val consumers = max(1, threads - producers)
+
+    val channelProducerConsumerBenchmarkWorker = ChannelProducerConsumerBenchmarkIterationMonteCarlo(
+            channelCreator, withSelect, producers, consumers, withBalancing,
+            dispatcherCreator, producers + consumers, APPROXIMATE_BATCH_SIZE)
+
+    val startTime = System.nanoTime()
+    channelProducerConsumerBenchmarkWorker.run()
+    val endTime = System.nanoTime()
+
+    val dispatcher = channelProducerConsumerBenchmarkWorker.dispatcher
+    if (dispatcher is Closeable) {
+        dispatcher.close()
+    }
+
+    return (endTime - startTime) / 1_000_000 // ms
+}
+
+class ChannelProducerConsumerBenchmarkIterationMonteCarlo(channelCreator: ChannelCreator,
```

Comment:
It would be nice to have much shorter class names. It is really hard to follow the code when reading `ChannelProducerConsumerBenchmarkIterationMonteCarlo(...ten params...) : ChannelProducerConsumerBenchmarkIteration`